rpcd/playbooks/roles/elasticsearch/files/reindex-liberty.py

```diff
@@ -0,0 +1,306 @@
+            master_name = index.replace(parsed_args.suffix, "")
+            counts[master_name] = _calc_pct(es,
+                                            parsed_args,
+                                            stats,
+                                            master_name,
+                                            index)
+
+    for master in counts.keys():
+        slave_index = master + parsed_args.suffix
+        master_count = counts[master][0]
+        slave_count = counts[master][1]
+        reindex_pct = counts[master][2]
+        if parsed_args.verbose:
+            print("Master Index: {0:15s} "
+                  "Slave Index: {1:15s} "
+                  "Master Count: {2:10d} "
+                  "Slave Count: {3:10d} "
+                  "Percentage Complete: {4:3.2f}%").format(master,
+                                                           slave_index,
+                                                           master_count,
+                                                           slave_count,
+                                                           reindex_pct)
+        total_count += master_count
+        done_count += slave_count
+    try:
+        pct_done = (float(done_count) / float(total_count)) * 100
+    except ZeroDivisionError:
+        pct_done = float(0.00)
+    if parsed_args.verbose:
+        print("Total Docments: {0:20d} "
+              "Reindexed Documents: {1:20d} "
+              "Percentage complete: {2:3.2f}%").format(total_count,
+                                                       done_count,
+                                                       pct_done)
+    else:
+        print("Percent complete: {0:3.2f}%").format(pct_done)
+    if pct_done == 100:
+        exit(code=0)
+    else:
+        return False
+
+
+def drop_legacy(es, parsed_args):
+    """Drop the legacy logstash-* indices."""
+    stats = get_stats(es)
+    for index in get_indices(es, parsed_args):
+        if _check_index(es, parsed_args, index, stats):
+            print("Dropping Legacy Index: {}").format(index)
+            if not parsed_args.dry_run:
+                es.indices.delete(index)
+
+
+def list_indices(es):
+    """List all indices."""
+    print(es.cat.indices(v=True))
+
+
+def batch(es, es_host, parsed_args):
+    """Batch Processing.
+
+    Batch processing, useful for performing re-indexing of old indices
+    prior to the actual upgrade.
+    """
+
+    today = date.today()
+    one_day = timedelta(days=1)
+    yesterday = today - one_day
+    indices = get_indices(es, parsed_args)
+    stats = get_stats(es)
+
+    if not parsed_args.start:
+        d_start = None  # Date object for start date
+        p_start = yesterday  # Previous first day for sort
+        for index in indices:
+            if 'logstash' in index:
+                s_date = _index_to_date(index)
+                if s_date < p_start:
+                    p_start = s_date
+        d_start = p_start
+    elif parsed_args.start:
```

Comment:
This and the `elif` in the following `if` statement should both probably be `else` since they're not testing other conditions, just that the `if` conditions already tested are false.